relay/network_graph/interests.py

```diff
@@ -0,0 +1,25 @@
+import time
+
+def calculate_interests(balance, rate, old_time, new_time):
+    return balance * (new_time - old_time) / (60 * 60 * 24 * 365) * rate / 100000
+
+def apply_interests(a, b, data):
+    """Applies the interest to the balance in data. The interpretation of data depends on whether a < b.
+    a and b play symmetrical roles"""
+    # apparently not, need to refactor
+    new_time = time.time()
+
+    # I assume interest_ba is the interest given by b to a. This might be wrong.
+    if a < b:
+        if data['balance_ab'] > 0:  # b owes a
+            interest = calculate_interests(data['balance_ab'], data['interest_ab'], data['m_time'], new_time)
+        else:  # a owes b
+            interest = calculate_interests(data['balance_ab'], data['interest_ba'], data['m_time'], new_time)
+    else:
```

Comment:
the if and then else block of the outer if statement are the same, which make me think there is an error.

Reply:
I do not think there is an error, the meaning of the data depends on the point of view of the data (a<b or b<a). But only one data is stored for both point of views. I believe this is confusing, which is why I kept both if and else statements even though the code is the same (easier to reason about for me).